WordPress/src/test/java/org/wordpress/android/util/wizard/WizardManagerTest.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.util.wizard
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.Observer
+import com.nhaarman.mockitokotlin2.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.ui.sitecreation.SiteCreationStep
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.DOMAINS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SEGMENTS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_PREVIEW
+
+private val STEPS = listOf(SEGMENTS, DOMAINS, SITE_PREVIEW)
+private val LAST_STEP_INDEX = STEPS.size - 1
+
+@RunWith(MockitoJUnitRunner::class)
+class WizardManagerTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
+
+    private lateinit var manager: WizardManager<SiteCreationStep>
+    @Mock private lateinit var navigatorLiveDataObserver: Observer<SiteCreationStep>
+
+    @Before
+    fun setUp() {
+        manager = WizardManager(STEPS)
+        manager.navigatorLiveData.observeForever(navigatorLiveDataObserver)
+    }
+
+    @Test
+    fun `showNextStep is propagated`() {
+        manager.showNextStep()
+        verify(navigatorLiveDataObserver).onChanged(STEPS[0])
```

Comment:
Similar to my earlier comment in `showNextStep increments currentStepIndex` test, I feel like it might be best to put this into a loop with valid indices to convey the intention of the code as well as validating more than one state. If we are keeping it as is, I'd at least propose to change the `STEPS[0]` to something like `any()` as otherwise this test looks to me like it's only validating the propagation after the initial state. If that was the intention, we could improve the naming of the test instead.

P.S: Sorry for so many alternatives. I wasn't sure of the intention of the test, so as I was typing I kept thinking of different suggestions.

Reply:
👍 I've just changed STEPS[0] to any() since my intention was to test that navigatorLiveDataObserver is called from within showNextStep(). I agree having STEPS[0] was a bit confusing.